Xamarin.Forms.Core/BindingExpression.cs

```diff
@@ -338,8 +338,7 @@ void SetupPart(TypeInfo sourceType, BindingExpressionPart part)
 				}
 			}
 			else
-				property = sourceType.GetDeclaredProperty(part.Content) ?? sourceType.BaseType?.GetProperty(part.Content);
-
+				property = sourceType.GetDeclaredProperty(part.Content) ?? sourceType.BaseType?.GetProperties().Where(pi => pi.Name == part.Content).FirstOrDefault();
```

Comment:
Is there any concern about which property gets returned?  I'm guessing GetProperties() isn't going to be consistent. But not really sure if there's much that can be done about that outside of possibly removing ours ahead of theirs

Reply:
GetProperty() is consistently returning the properties from the type, then from the parent. that's the only kind of consistency we care about here

Reply:
> GetProperty() is consistently returning the properties from the type, then from the parent.

I feel like that might still differ system to system even though it's what you're seeing.
Something that enforces that hierarchy seems safer

For example something like this:
https://stackoverflow.com/a/33481994/953734

```C#
property = sourceType.GetDeclaredProperty(part.Content);
if(property == null)
{
	var properties = sourceType.BaseType?.GetProperties().Where(pi => pi.Name == part.Content).ToList();

	property = properties?.FirstOrDefault(x => x.DeclaringType.Name != "BindableObject")  ?? properties?.FirstOrDefault();
}
```

Reply:
@PureWeen the hierarchy is somehow preserved already (first look in the current, then in parents). I don't understand how your proposed fix is supposed to be more bulletproof than the original solution. Also, it's unlikely that the source will ever derive from BindableObject.

What's still strange to me, is that `GetProperties()` is not, according to the doc, supposed to use the `FlattenHierarchy` flag, but still does...

Reply:
@PureWeen order is now guaranteed

Reply:
Yea my suggestion was a bit off :-/  The idea was more that it would leave out the base type and check the derived type first but the way I went about it wasn't very well put together.

I did a few tests with GetProperties and shadowing and from what I could tell GetProperties only returned the most derived version and the initial declaration. Which is where I was going with that fix. Like if you have a middle type that also shadowed that wasn't being returned by GetProperties.

I like the idea here but I don't think this compiles for netstandard 1.0 

What about using *IsAssignableFrom*

```C#
	internal class PropertyInfoDeclaringType : IComparer<PropertyInfo>
	{
		public int Compare(PropertyInfo x, PropertyInfo y)
		{
			if (x == y)
				return 0;

			if (x.DeclaringType.IsAssignableFrom(y.DeclaringType))
				return 1;

			return -1;
		}
	}


property = 
					sourceType.BaseType.GetProperties().Where(pi => pi.Name == part.Content)
					.OrderBy(x=> x, new PropertyInfoDeclaringType())
					.FirstOrDefault();
```